### PR TITLE
Expand rows

### PR DESCRIPTION
**Pull request by nimam-bentley**
- report problems table expandable by grouped categories/issues
- entire row is clickable 
- sync time/files moved from banner tile to inline text

### CODE REVIEW

File: src/components/ProblemsTable.tsx [hunk 1]
```diff
@@ -92,7 +91,38 @@ export const ProblemsTable = ({
         return context?.focusedIssues.some((issue) => bannerLevel === issue);
       });
 
-    return reports;
+    // **FIX referenced in own type annotation
```
Comment (nimam-bentley) [inline src/components/ProblemsTable.tsx:94]: please mention if any suggestions for improving type annotations

File: src/components/ProblemsTable.tsx [hunk 2]
```diff
@@ -244,6 +284,27 @@ export const ProblemsTable = ({
...
     <Table
+      onRowClick={(event: React.MouseEvent<Element, MouseEvent>, row: Row<TableRow | TableRow['subRows']>) => {
+        const element = event.target as HTMLTableRowElement;
+        element.parentElement?.classList.add('active');
+        context?.setCurrentAuditInfo({
+          ...row.original,
+          fileName: row.original.filename ?? getFileNameFromId(row.original.fileId),
+        });
+
+        // event listener added as component does not have an 'outside clicked' handler
```
Comment (nimam-bentley) [inline src/components/ProblemsTable.tsx:294]: please mention if any suggestions for more suiting implementation, in terms of React or component library

File: src/components/ProblemsTable.tsx [hunk 3]
```diff
@@ -92,7 +91,37 @@ export const ProblemsTable = ({
         return context?.focusedIssues.some((issue) => bannerLevel === issue);
       });
 
-    return reports;
+    // **FIX referenced in own type annotation
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const expandableReports: any = {};
```
Comment (Mindaugas-Adomavicius) [inline src/components/ProblemsTable.tsx:96]: where any is used as a type, can't we self-define types which would for this instance define expandableReport

File: src/components/ProblemsTable.tsx [hunk 4]
```diff
@@ -92,7 +91,37 @@ export const ProblemsTable = ({
...
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (Mindaugas-Adomavicius) [inline src/components/ProblemsTable.tsx:97]: can't we move this new way to return a report to a separate function or even file if needed?

File: src/components/ProblemsTable.tsx [hunk 5]
```diff
@@ -135,18 +167,9 @@ export const ProblemsTable = ({
...
-                  ...row.row.original,
-                  fileName: row.row.original.fileName ?? getFileNameFromId(row.row.original.fileId),
-                });
-              }}
-            >
-              {row.value}
-            </div>
+          Cell: (row: CellProps<TableRow | TableRow['subRows']>) => (
+            // Hide issue if a subrow
+            <div>{!Object.hasOwn(row.row.original, 'subRows') ? '' : row.value}</div>
```
Comment (Mindaugas-Adomavicius) [inline src/components/ProblemsTable.tsx:172]: can't we reverse the check like Object.hasOwn(row.row.original, 'subRows') ? row.value :  '' ?

File: src/components/ProblemsTable.tsx [hunk 6]
```diff
@@ -244,6 +283,27 @@ export const ProblemsTable = ({
 
   return (
     <Table
+      onRowClick={(event: React.MouseEvent<Element, MouseEvent>, row: Row<TableRow | TableRow['subRows']>) => {
+        const element = event.target as HTMLTableRowElement;
```
Comment (Mindaugas-Adomavicius) [inline src/components/ProblemsTable.tsx:287]: move the whole onRowClick function to separate function as writing everything in line decreases the readability

File: src/components/Report.tsx [hunk 7]
```diff
@@ -33,7 +33,6 @@ type AuditInfo = Partial<{
   type: string;
   fileName: string;
   filePath: string;
-  impactedWorkflows: string[];
```
Comment (Mindaugas-Adomavicius) [inline src/components/Report.tsx:36]: why there was a removal of impactedWorkflows? Need more context

File: src/components/ProblemsTable.tsx [hunk 8]
```diff
@@ -244,6 +283,27 @@ export const ProblemsTable = ({
...
+        element.parentElement?.classList.add('active');
+        context?.setCurrentAuditInfo({
+          ...row.original,
+          fileName: row.original.filename ?? getFileNameFromId(row.original.fileId),
+        });
+
+        // event listener added as component does not have an 'outside clicked' handler
+        // and cannot wrap individual row in a custom react hook
+        document.addEventListener(
```
Comment (Mindaugas-Adomavicius) [inline src/components/ProblemsTable.tsx:296]: can't we use useState and useEffect for a row expansion? You could keep the row state, currently active row in states and open, close it with useEffect?

File: src/components/ReportTimestamp.tsx [hunk 9]
```diff
@@ -74,20 +74,14 @@ export const ReportTimestamp = ({
...
-        <Text variant='small'>{displayStrings.syncTime}</Text>
-      </BannerTile>
-      <BannerTile icon={<SvgDocument />}>
-        <Text variant='title' style={{ fontWeight: 'bold' }}>
-          {allFilesProcessed.length}
+    <div className='isr-timestamp-container'>
+      <span>
+        <Text>
+          {' '}
+          <SvgClock /> {date} {time} {displayStrings.syncTime} | <SvgDocument /> {allFilesProcessed.length}{' '}
```
[the remark this answers is not in the document]
Comment (nimam-bentley) [reply src/components/ReportTimestamp.tsx:81]: yes

File: src/components/Report.tsx (hunk 7 above)
Comment (nimam-bentley) [reply src/components/Report.tsx:36]: impacted workflow was removed in [recent PR](PR), this seems to have just been forgotten to be removed

**Review by Mindaugas-Adomavicius — APPROVED** [review]

File: src/components/ProblemsTable.tsx [hunk 10]
```diff
@@ -242,8 +288,32 @@ export const ProblemsTable = ({
     []
   );
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onRowClick = (event: React.MouseEvent<Element, MouseEvent>, row: Row<Record<string, any>>): void => {
```
Comment (Guillar1) [inline src/components/ProblemsTable.tsx:292]: This should be memoized with useCallback.

**Review by Aidas-Vaiksnoras — APPROVED** [review]

File: package.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@itwin/synchronization-report-react",
-  "version": "2.0.0",
+  "version": "2.0.0-dev",
```
Comment (aruniverse) [inline package.json:3]: Dev what?

Should be 2.0.1-dev.0
Or 2.1.0-dev.0

**Pull request merged**